### DISCUSSION TITLE
Produce reference docs for FastAPI

### DISCUSSION THREAD

**Discussion by corydodt** [fastapi/fastapi#6822] [Questions] [answered]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [ ] After submitting this, I commit to:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * Or, I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Description

* FastAPI's docs, while very good, do not have a reference view to the internal API. What are the arguments that a Response object takes? What are the arguments that a Query object takes? etc. A standard, autogenerated reference doc is needed! Now that I have a basic understanding of the things this framework can do, 90% of my questions can be answered by an online reference.

### The solution you would like

Autogenerate reference docs and publish them with the rest of the docs, linked from the docs page.

One of this project's primary focuses is document generation for framework users. Seems like it should put that philosophy into practice for itself.

### Describe alternatives you've considered

- Build my own documentation for this framework?
- Search the source code? (what I've resorted to doing most of the time)

**Comment by Kludex** [comment]
We already have this PR and @tiangolo's opinion there: https://github.com/tiangolo/fastapi/pull/2275

**Comment by ycd** [comment, marked as answer]
There are a lot of duplicate issues, see:
#1588
#845
#804

We already have a PR(#2275) about it which is opened by @Kludex .

**Comment by corydodt** [comment]
Closing as there are dupes. (I'm going to blame github search for this, as I did search for the terms in this case, but came up empty-handed.)

I find myself puzzled by @tiangolo rejecting the existence of API docs because of planned API changes in the future. When those API changes land, publish new docs, just as you would with any other change. It's **much, much easier** to change and republish autogenerated docs than to fix the dozens of prose docs that already exist, so I don't see how that tracks.

**Comment by tiangolo** [comment]
Thanks for the help here everyone! 👏  🙇 

Thanks for reporting back and closing the issue 👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
